### ISSUE TITLE
Fix `remove unused variable` does nothing in loops

### ISSUE THREAD

**Issue by neonaot** [intellij-rust/intellij-rust#8941]
## Environment

* **IntelliJ Rust plugin version:** 0.4.172.4663-222-nightly
* **Rust toolchain version:** 1.63.0-nightly (546c826f0 2022-06-12) x86_64-apple-darwin
* **IDE name and version:** CLion 2022.2 EAP (CL-222.2680.4)
* **Operating system:** macOS 11.4
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.cargo.emulate.terminal, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description

https://user-images.githubusercontent.com/49211026/173882342-13aed6a7-09c9-4436-86ae-73dfa93a0220.mov

More natural example with same warning:
```rust
    for i in 0..5{
        println!("hello");
    }
```    

## Steps to reproduce
```rust
fn main() {
    let s: &str = "Hello, World!";

    for iterator in s.chars() {
        println!("hello");
    }
}
```
## Expected 

```rust
fn main() {
    let s: &str = "Hello, World!";

    for _ in s.chars() {
        println!("hello");
    }
}
```
or don't show this warning in the loops.
